mne/decoding/search_light.py

```diff
@@ -228,7 +228,16 @@ def score(self, X, y):
             Score for each estimator / data slice couple.
         """
         from sklearn.metrics import make_scorer, get_scorer
+
+        if len(np.unique(y)) <= 2:
+            from sklearn.preprocessing import LabelEncoder
+                                     
+            le = LabelEncoder()
+            y = le.fit_transform(y)
+
         self._check_Xy(X)
+        le = LabelEncoder()
+        y = le.fit_transform(y)
```

Comment:
@kingjr I think the current change satisfies the behavior you have mentioned. Labels are only converted to `[0, 1]` while using score function. I have now removed this redundant `LabelEncoder`.